### DISCUSSION TITLE
Allow new user to sign-up

### DISCUSSION THREAD

**Discussion by cywhale** [biigle/core#338] [Q&A] [answered]
I had built a Biigle server and want to add a Sign-up form for new users to apply account. 
Seems Biigle core already had this function/module #173
But sorry I cannot find this function or module (neither from admin documentation) to add to my build.
Is there any way to add auto-signup function or example to implement it/ or Oauth login? Thanks.

**Comment by mzur** [comment, marked as answer]
You can enable the sign-up form by setting the [`BIIGLE_USER_REGISTRATION`](https://github.com/biigle/core/blob/2547db96819affd90bcc66ddb965b62dd6b00a57/config/biigle.php#L28) environment variable to `true` (in the [`.env`](https://github.com/biigle/biigle/blob/master/build/.env.example) file of your production configuration).

Adding other authentication methods is much more complex and not supported out of the box. There are Laravel packages for alternative authentication methods, though, so you could implement it on your own. We already experimented with LDAP authentication for the deployment in a research institute but this was not finalized and is not officially supported.

**Reply by cywhale** [reply]
Got it. It's neat. Thanks for your kind help.
